Content.Server/Flash/FlashSystem.cs

```diff
@@ -196,11 +196,14 @@ private void OnFlashExamined(EntityUid uid, FlashComponent comp, ExaminedEvent a
         private void OnInventoryFlashAttempt(EntityUid uid, InventoryComponent component, FlashAttemptEvent args)
         {
             // Forward the event to a worn helmet, if one is equipped.
-            if (_inventorySystem.TryGetSlotEntity(uid, "head", out var maskSlotEntity, component))
-                RaiseLocalEvent(maskSlotEntity.Value, args);
+            if (_inventorySystem.TryGetSlotEntity(uid, "head", out var headSlotEntity, component))
+                RaiseLocalEvent(headSlotEntity.Value, args);
             // Forward the event to the glasses, if any.
             if(!args.Cancelled && _inventorySystem.TryGetSlotEntity(uid, "eyes", out var eyeSlotEntity, component))
                 RaiseLocalEvent(eyeSlotEntity.Value, args);
+            // Forward the event to the mask, if any.
+            if (!args.Cancelled && _inventorySystem.TryGetSlotEntity(uid, "mask", out var maskSlotEntity, component))
+                RaiseLocalEvent(maskSlotEntity.Value, args);
```

Comment:
I don't know how i feel about tacking on more specific slot checks. the best way to implement this would be the enumerate over the entire inventory, check for the FlashImmunityComponent, then raise the event on that piece. The less needless hard-coding of slot ids, the better.